Notifications API actions/maxActions and friends (#44184)
mdn/content: files/en-us/web/api/notifications_api/index.md

{{DefaultAPISidebar("Web Notifications")}}{{securecontext_header}} {{AvailableInWorkers}}

The Notifications API allows web pages to control the display of system notifications to the end user.

## Concepts and usage

A web notification is a message box used to inform users when events occur on web apps. Web notifications are rendered by the operating system's native notification system, making them display identically to notifications from any other app on the platform.

> Web notifications are displayed by the underlying OS; they are therefore outside the top-level browsing context viewport, and can be shown even when the user has switched tabs or moved to a different app.

Reviewer: ```suggestion
Because the underlying OS renders web notifications, they are outside the top-level browsing context viewport, and can be shown even when the user has switched tabs or moved to a different app.
```

Re-reading the whole paragraph together, it sounds a bit weird to say "Web notifications are rendered..." and then "Web notifications are displayed..." immediately in the next line.

Author: Yeah, that's why I had it this way originally :-). Thanks.

{{DefaultAPISidebar("Web Notifications")}}{{securecontext_header}} {{AvailableInWorkers}}

The Notifications API allows web pages to control the display of system notifications to the end user.

## Concepts and usage

> A notification is a system-level UI element rendered by the operating system's native notification system, making it identical to notifications from any other app on the platform.

Reviewer: ```suggestion
A web notification is a message box used to inform users when events occur on web apps. Web notifications are rendered by the operating system's native notification system, making them display identically to notifications from any other app on the platform.
```

And then delete the next line.

I thought the description should start with what web notifications are and what their purpose is. Also, and this is pretty picky — they are not identical to other apps' notifications; they are just displayed identically.

> It is used to inform users of events.

Author: ```suggestion
```

> Because it is a system-level element, it is outside the top-level browsing context viewport, and can be displayed even when the user has switched tabs or moved to a different app.

Reviewer: ```suggestion
Web notifications are displayed by the underlying OS; they are therefore outside the top-level browsing context viewport, and can be shown even when the user has switched tabs or moved to a different app.
```

There are two types of notifications: persistent and non-persistent.

> Persistent notifications can run on mobile devices, are created through service workers, and can exist after their originating page has been closed.

Reviewer: Present these as bullet points, to make the info a little easier to parse?

Author: I did that in the next section. This was all about having a succint introduction. However in retrospect this is too much duplication so I think I'll remove it all.

Reviewer: Much better

By contrast, non-persistent notifications can only be used in desktop operating systems and have a lifetime bounded by the {{domxref("Notification")}} instance — and hence their originating page.

> Both types require user-permission before a notification can be created.

Reviewer: ```suggestion
Both types require the user's permission before a notification can be created.
```

> 

Author: ```suggestion
```

### Persistent and non-persistent notifications

The Notifications API supports two types of notifications:

- **Non-persistent notifications** are created in a browsing context, such as a web page or tab.
  Their lifetime is tied to the lifetime of the page — if the page is closed, the notification can no longer be interacted with.

  They are created using the {{domxref("Notification.Notification","Notification()")}} constructor and fire events such as {{domxref("Notification/click_event", "click")}} directly on the `Notification` instance.

- **Persistent notifications** are created from a service worker, and can remain interactive beyond the lifetime of an individual page.

>   They are created using {{domxref("ServiceWorkerRegistration.showNotification()")}} from a service worker and fire {{domxref("ServiceWorkerGlobalScope/notificationclick_event", "notificationclick")}} and {{domxref("ServiceWorkerGlobalScope/notificationclose_event", "notificationclose")}} events on the {{domxref("ServiceWorkerGlobalScope")}}.

Reviewer: ```suggestion
  They are created by calling {{domxref("ServiceWorkerRegistration.showNotification()")}} from inside a service worker and fire {{domxref("ServiceWorkerGlobalScope/notificationclick_event", "notificationclick")}} and {{domxref("ServiceWorkerGlobalScope/notificationclose_event", "notificationclose")}} events on the {{domxref("ServiceWorkerGlobalScope")}}.
```
